samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -301,7 +301,7 @@ private void assignTasksToContainers(int[] taskCountPerContainer, List<String> t
         containerTaskModels.put(model.getTaskName(), model);
       }
       containerModels.add(
-          new ContainerModel(container.containerId, Integer.valueOf(container.containerId), containerTaskModels));
```

Comment:
Since we're getting rid of containerId field from ContainerModel, just for the sake of consistency in other internal classes like `TaskGroup`, can we rename instance-fields, variables like `containerId` to `processorId`?  

Otherwise, after this change it will be hard for a newcomer without context to understand why `processorId` and `containerId` are used interchangeably in the grouper and it's internal classes.

 If you agree to this, then please change it.

Reply:
I'd prefer to not call fields/getters just "id". We have so many that it can be confusing to figure out which one is intended at call sites.

+1 for renaming fields and variables. We can do a general s/Container/Processor renaming later.

Reply:
> What do you think about naming the field id in ContainerModel and having a getId method.

Yes that sounds like a good idea to hide the complexity. At-least it would prevent us from exposing `getProcessorId` method in `ContainerModel`(`getProcessorId` seems to be misnomer to expose from `ContainerModel`).

Reply:
Discussed with @prateekm offline. Since callers will need to use containerModel.getId, it still is clear enough that it is the container id. Callers will need to ensure that they clearly name their variables so that they don't get mixed up by the multiple kinds of ids.